### ISSUE TITLE
Request: Make app available to F-Droid

### ISSUE THREAD

**Issue by huuhaa** [ruuvi/com.ruuvi.station#1027]
It would be great if you could make app available to f-droid. I know it probably requires some changes to make it happen, mostly removing trackers etc. from f-droid build, but still at the end I think it would be worth it.

I hope you could at least comment what are you thoughts about this.

**Comment by laurijamsa** [comment]
Thanks for the suggestion.

As far as I remember, no one has requested this before and therefore we're of course a bit unsure if it would be actually worth it :) 

Please also note that Ruuvi Station automated builds are already available on GitHub as APKs.

**Comment by laurijamsa** [comment]
Thanks for the additional opinions and points. We can keep the issue open for now and we will also discuss about this internally with our team.

**Comment by andreevdenis** [comment]
@huuhaa Hey! Is screenshot you provided based on Ruuvi Station app analysis? What did you use to get it?

**Comment by huuhaa** [comment]
@andreevdenis Hi! Screenshot is from Ruuvi Station"report", for that I used TrackerControl:

https://trackercontrol.org/

At website there are download links, but do note that you can't get full fuctions at Play Store version, cause Google don't want you to limit tracking. So either use version from github (should have notifications for updates), or get from F-Droid (I use F-Droid version personally).

Usage of app is easy, basicly:
- Install Tracker Control
- Open it first time and setup local VPN
- Don't select block essentials by default if this is new to you, you would have to finetune app by app and allow some trackers for some apps to work, so easier to allow essentials by default..
-  Use your apps like normally and check from TrackerControl what connections it has blocked/allowed. It also offers trafficlog feature if intrested to see more detailed information

But that's about it "shortly", I stop now cause bit offtopic, but still related though.

**Comment by IzzySoft** [comment]
> I personally don't know what changes would be required to get app even on IzzyOnDroid repo...

As you've pinged me, I've taken a look at the APK. And unfortunately you are correct, that wouldn't even meet the [inclusion criteria with my repo](https://gitlab.com/IzzyOnDroid/repo/#what-are-the-requirements-an-app-must-meet-to-be-included-with-the-repo):

> it preferably has no proprietary components. While some of them might be acceptable, trackers (e.g. ads, analytics) are tolerated at best but only if there are no more than two such modules.

```
Offending libs:
---------------
* Crashlytics (/com/crashlytics): NonFreeDep,Tracking
* Firebase Data Transport (/com/google/android/datatransport): NonFreeNet
* Google Cloud Messaging (/com/google/android/gms/cloudmessaging): NonFreeDep,NonFreeNet
* Play Services Places Placereport (/com/google/android/gms/places_placereport): NonFreeDep,NonFreeNet
* Google Mobile Services (/com/google/android/gms): NonFreeDep
* Google Play Core (/com/google/android/play/core): NonFreeNet,NonFreeDep
* Cloud Audit Logs (/com/google/cloud/audit): Tracking
* Google Cloud Logging Client for Java (/com/google/cloud/logging): NonFreeNet
* Firebase (/com/google/firebase): NonFreeNet,NonFreeDep
* Firebase Analytics (/com/google/firebase/analytics): NonFreeDep,Tracking
* Firebase In-App Messaging SDK (/com/google/firebase/inappmessaging): NonFreeDep,NonFreeNet
* Firebase Installations (/com/google/firebase/installations): NonFreeNet

12 offenders.
```

I need to update the details there. For about at least a year or two, the overall number of proprietary components is limited to 5. While that's not "put into stone" (I e.g. tend to "group" some things, like counting all Firebase libs (except analytics) as one), 12 is definitely way too much. A foss/floss flavor could help out here, having the "real bad stuff" (Crashlytics, Firebase Analytics, Cloud Audit) removed – which then would probably also implicitly get rid of some more which are just "dragged in" as dependencies.

An alternative approach benefiting all people using your app would be to either remove those things altogether – or replace them with something F/LOSS and more privacy-friendly:

* for push notifications, take a look at @unifiedpush (you could even include their FCM distributor for your PlayStore releases, so folks there wouldn't note a difference – while leaving the F/LOSS folks [their own choice](https://android.izzysoft.de/applists/category/named/network_push))
* for analytics, see my snippet on [acceptable analytics](https://gitlab.com/-/snippets/1896503#acceptable-analytics). Hopefully something from that list would be fine for you as well
* should you use other Firebase functionality, take a look at [appwrite](https://github.com/appwrite/appwrite) (fitting your own license well, as it uses the same) or [Supabase](https://supabase.com/) (uses Apache-2.0)

The only remaining candidate then would be PlacesReport – which I have no idea a) what you need it for and thus b) what could replace it.

**TL;DR:** if you can get the above list-of-offenders down to below 6, I could accept your app into my repo – while you fight down the rest to be eligible for F-Droid.org. Deal?

**Comment by IzzySoft** [comment]
@huuhaa you know that you can find [TrackerControl in my repo](https://apt.izzysoft.de/fdroid/index/apk/net.kollnig.missioncontrol) – that is, the full version you mentioned being available at Github (and for completeness, here's the link to the dedicated [F-Droid version](https://f-droid.org/packages/net.kollnig.missioncontrol.fdroid/))? That said, Konrad and his team doing a great job with that at the Oxford university :star_struck:

**Comment by juleskers** [comment]
Just a +1 from me.
I'm a happy Ruuvi user (for per-room temperature monitoring, including child's bedroom).

I would love a privacy-friendly, de-googled Ruuvi app on my de-googled fairphone ( [e/OS](https://e.foundation/).  
(I realise this probably self-identifies me as already fairly zealous).

Another angle to consider is that all those (American) tracking companies make for a legal headache under EU GDPR. 
Can Ruuvi honestly say they checked if all those services and dependencies are covered by up-to-date data sharing agreements? (Safe Harbour and it's descendants keep getting stricken down by courts) 

Also a compliment to the Ruuvi team: we wouldn't even be having this discussion if you didn't do such great open source work!
Just the fact that you publish this repo is 80%. Please allow us zealots to assist you in the final 20% towards true F/LOSS. ❤️

**Comment by huuhaa** [comment]
> As you've pinged me, I've taken a look at the APK. And unfortunately you are correct, that wouldn't even meet the [inclusion criteria with my repo](https://gitlab.com/IzzyOnDroid/repo/#what-are-the-requirements-an-app-must-meet-to-be-included-with-the-repo):

Sorry for extremely long delay in answering (computer broke, noticed your message, but didn't answer using mobile during that.. gor computer fixed and had forgotten this..).

Yep I pinged you cause was thinking you could point out what changes would needed to do to make app either f-droid official repo or your repo, like you did, thanks for that!

About TrackerControl, yes I know it's also awailable at your repo, thanks for that!

And what coes to RuuviStation itself.. Seems hopes are not high, but like saying goes "hope dies last". So I still hope Ruuvi team will consider this.

**Comment by IzzySoft** [comment]
> So I still hope Ruuvi team will consider this.

So do I – but it is more than 3 months now since I mentioned viable approaches, and there's still no word from the Ruuvi team. Hope is still there, but expectations got down quite a bit. A clear word would help.

**Comment by juleskers** [comment]
Let's not be too harsh, @IzzySoft, the amount of libraries that Ruuvi would need to replace/remove represent quote a massive shift in their development workflow.

We personally may not like what those libraries do, but they definitely have value for commercial entities. 
From outside we cannot judge how critical those are to their bugfixing and general development process, let alone how essential the metrics are for their business.

Give ruuvi some credit for open sourcing this Repo at all, rather than complaining for not being all-the-way f-droid-grade open source.

**Comment by IzzySoft** [comment]
Not being harsh here (and never intended to). By "a clear word" I meant, for now, just a declaration of intent (from the hopefully-not "no we won't do it" via "we look into it" and "can be partly done" to the ideal "we'll address this ASAP"). Im not disappointed it wasn't yet solved (here I fully agree with you that to solve ALL of them will be quite some work), but only that the issue seems to not get much attention. The last note from the team was given by @laurijamsa back in June that they will discuss it in the team (fair enough!) – but we never learned the outcome.

And yes, definitely: credit where it is due, I'm fully with you on that @juleskers! And if I voice my disappointment, it's not meant as complaining (I'm not even affected myself, so why should I complain). Rather count it as means of motivation (like "ugh, yes, we totally forgot to update you folks – we were quite busy with other tasks" – and yes, I know first hand what "being busy with too many tasks" means).

So a heads-up to the team: we didn't give up hope on you and really appreciate your work :wink:

**Comment by laurijamsa** [comment]
Sorry guys for not updating this issue sooner and thank you for the kind words. 

According to the above discussion, to make the app fully floss compatible, we would need to use time and resources by studying, implementing and testing alternatives for crashlytics and messaging dependencies. The fact is that have our hands full with feature development and we have to prioritise. Therefore we have decided that we cannot promise that we will make the app to meet the above mentioned criteria. But we will definitely anyways double check the tracker list you provided above and see if there is something that we could stop using without losing the features we need.

Leaving this issue open for now and moving it to todo (to double check the trackers). After that we will decide it it's time to close this or what to do.

**Comment by IzzySoft** [comment]
Thanks @laurijamsa – we cannot ask for more! As a pointer: you could use a [build flavor](https://developer.android.com/build/build-variants) to test these things out. Maybe even a `foss` one with most (or all?) of the offenders removed/replaced. I could then pick that and provide it via [my repo](https://apt.izzysoft.de/fdroid), having a special note you name in the description. That would offer you a base of testers (those using my repo) while the `gplay` variant (the current setup) would remain untouched. A clear win-win then: our privacy folks get their "foss build" – and you get your testers for that. Again, just a suggestion – and no pressure :wink:

**Comment by markoaamunkajo** [comment]
@andreevdenis please check the comment from @laurijamsa to check trackers etc. as investigation and possibly easy fix.

**Comment by markoaamunkajo** [comment]
_No signs of unexpected trackers were found in our codebase and application that detects trackers only show me Google related hosts._


@laurijamsa

**Comment by IzzySoft** [comment]
> No signs of unexpected trackers were found in our codebase and application that detects trackers only show me Google related hosts.

```
$ scanapk Ruuvi.Station.2.5.6.apk

Offending libs:
---------------
* Crashlytics (/com/crashlytics): NonFreeComp,Tracking
* Firebase Data Transport (/com/google/android/datatransport): NonFreeNet
* Google Cloud Messaging (/com/google/android/gms/cloudmessaging): NonFreeComp,NonFreeNet
* Play Services Places Placereport (/com/google/android/gms/places_placereport): NonFreeComp,NonFreeNet
* Google Mobile Services (/com/google/android/gms): NonFreeComp
* Google Play Core (/com/google/android/play/core): NonFreeNet,NonFreeComp
* Cloud Audit Logs (/com/google/cloud/audit): Tracking
* Google Cloud Logging Client for Java (/com/google/cloud/logging): NonFreeNet
* Firebase (/com/google/firebase): NonFreeNet,NonFreeComp
* Firebase Analytics (/com/google/firebase/analytics): NonFreeComp,Tracking
* Firebase In-App Messaging SDK (/com/google/firebase/inappmessaging): NonFreeDep,NonFreeNet
* Firebase Installations (/com/google/firebase/installations): NonFreeNet

12 offenders.

```

7 non-free components, 3 marked for tracking, 8 for requiring non-free network services. But right, you wrote "no signs of **unexpected** trackers, so those are **expected** ones. Which unfortunately makes your app not suited for listing at F-Droid.org or my repo – and basically not fully FOSS either (as those non-free components are NOT open source).

We'll accept your decision of course, but it's a bit disappointing. Those who are more privacy-focused will thus be unable to use your app. Not complaining, just stating the facts. Thanks for investigating, though!
